core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.kt

```diff
@@ -42,42 +67,141 @@ abstract class CoreMainActivity : BaseActivity(), WebViewProvider {
+    drawerToggle.isDrawerIndicatorEnabled = false
+    drawerContainerLayout.setDrawerLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED)
+  }
+
+  open fun onNavigationItemSelected(item: MenuItem): Boolean {
+    when (item.itemId) {
+      R.id.menu_support_kiwix -> openSupportKiwixExternalLink()
+      R.id.menu_settings -> openSettingsActivity()
+      R.id.menu_help -> start<HelpActivity>()
+      R.id.menu_history -> openHistoryActivity()
+      R.id.menu_bookmarks_list -> openBookmarksActivity()
+      else -> return false
+    }
+    return true
+  }
+
+  private fun navigationDrawerIsOpen(): Boolean =
+    drawerContainerLayout.isDrawerOpen(drawerNavView)
+
+  fun closeNavigationDrawer() {
+    drawerContainerLayout.closeDrawer(drawerNavView)
+  }
+
+  private fun openSupportKiwixExternalLink() {
+    externalLinkOpener.openExternalUrl(KIWIX_SUPPORT_URL.toUri().browserIntent())
+  }
+
   override fun onBackPressed() {
-    supportFragmentManager.fragments.filterIsInstance<BaseFragmentActivityExtensions>().forEach {
-      if (it.onBackPressed(this) == ShouldCall) {
+    if (navigationDrawerIsOpen()) {
+      closeNavigationDrawer()
+      return
+    }
+    if (activeFragments().filterIsInstance<FragmentActivityExtensions>().isEmpty()) {
+      return super.onBackPressed()
+    }
+    activeFragments().filterIsInstance<FragmentActivityExtensions>().forEach {
+      if (it.onBackPressed(this) == FragmentActivityExtensions.Super.ShouldCall) {
         super.onBackPressed()
       }
     }
   }
 
-  override fun onNewIntent(intent: Intent) {
-    super.onNewIntent(intent)
-    supportFragmentManager.fragments.filterIsInstance<BaseFragmentActivityExtensions>().forEach {
-      it.onNewIntent(intent, this)
+  override fun onCreateOptionsMenu(menu: Menu): Boolean {
+
+    if (activeFragments().filterIsInstance<FragmentActivityExtensions>().isEmpty()) {
+      return super.onCreateOptionsMenu(menu)
+    }
+    var returnValue = true
+    activeFragments().filterIsInstance<FragmentActivityExtensions>().forEach {
+      if (it.onCreateOptionsMenu(menu, this) == FragmentActivityExtensions.Super.ShouldCall) {
+        returnValue = super.onCreateOptionsMenu(menu)
+      }
     }
+    return returnValue
   }
 
-  override fun getCurrentWebView(): KiwixWebView? {
-    return supportFragmentManager.fragments.filterIsInstance<WebViewProvider>().first()
-      .getCurrentWebView()
+  private fun activeFragments(): MutableList<Fragment> =
+    supportFragmentManager.fragments
+
+  fun navigate(action: NavDirections) {
+    navController.navigate(action)
   }
+
+  fun navigate(fragmentId: Int) {
+    navController.navigate(fragmentId)
+  }
+
+  fun navigate(fragmentId: Int, bundle: Bundle) {
+    navController.navigate(fragmentId, bundle)
+  }
+
+  abstract fun openSettingsActivity()
+
+  private fun openHistoryActivity() {
+    navigate(historyFragmentResId)
+    handleDrawerOnNavigation()
+  }
+
+  private fun openBookmarksActivity() {
```

Comment:
Opening a fragment needs to be followed by disabling the navigation drawer which you are correctly implementing using `handleDrawerOnNavigation()`. But what if that fragment is not opened from this activity the navDrawer won't be disbaled. 
For example if I bookmark a page from reader and open the bookmarks page using the snackbar action the navigation drawer won't be disabled.
This can be handeled using the `onOptionsItemSelected` on the fragment itself , if I'm not wrong the bookmarks fragment is using the `PageFragment` so making some changes there would fix this issue.

Reply:
Good catch! Will fix.